datafusion/catalog-listing/src/table.rs

```diff
@@ -578,6 +578,11 @@ impl TableProvider for ListingTable {
         let keep_partition_by_columns =
             state.config_options().execution.keep_partition_by_columns;
 
+        // Invalidate cache entries for this table if they exist
+        if let Some(lfc) = state.runtime_env().cache_manager.get_list_files_cache() {
+            let _ = lfc.remove(table_path.prefix());
+        }
```

Comment:
For the initial implementation I chose to use the simple option of invalidating a table's cache entries on `INSERT`. Since the default TTL is infinite any `DataSink` that only supports writing new files (as opposed to appending) would never pick up inserts without some action being taken here.

It would be very nice to handle this more elegantly in the future.

Reply:
makes sense. Thank you